--- a/src/internal/network_conf_function.c
+++ b/src/internal/network_conf_function.c
@@ -28 +28 @@
-int get_services_str(char *line, FILE *f, int *indexPtr)
+char *get_services_str(char *line, FILE *f, int *indexPtr)
@@ -36 +36 @@
-		return 1;
+		return line;
--- a/src/internal/network_conf_function.h
+++ b/src/internal/network_conf_function.h
@@ -21 +21 @@
-int get_services_str(char *line, FILE *f, int *indexPtr);
+char *get_services_str(char *line, FILE *f, int *indexPtr);
--- a/src/network/lookup_name.c
+++ b/src/network/lookup_name.c
@@ -62 +62 @@
-static inline int get_hosts_str(char *line, int length, FILE *f, int *i)
+static inline char *get_hosts_str(char *line, int length, FILE *f, int *i)
@@ -78 +78 @@
-		return 1;
+		return line;
--- a/src/network/getnameinfo.c
+++ b/src/network/getnameinfo.c
@@ -52 +52 @@
-static inline int get_hosts_str(char *line, int length, FILE *f, int *i)
+static inline char *get_hosts_str(char *line, int length, FILE *f, int *i)
@@ -60 +60 @@
-		return 1;
+		return line;
--- a/src/network/getaddrinfo.c
+++ b/src/network/getaddrinfo.c
@@ -59 +59 @@
-		g_recursiveKey = NULL;
+		g_recursiveKey = 0;
--- a/ldso/cfi.c
+++ b/ldso/cfi.c
@@ -206 +206 @@
-    uintptr_t addr = func_ptr;
+    uintptr_t addr = (uintptr_t)func_ptr;
@@ -364 +364 @@
-    fill_shadow_value_to_shadow(dso->map, dso->map + dso->map_len, 0, sv_invalid);
+    fill_shadow_value_to_shadow((uintptr_t)dso->map, (uintptr_t)dso->map + dso->map_len, 0, sv_invalid);
@@ -421 +421 @@
-            if (fill_shadow_value_to_shadow(p->map, p->map + p->map_len, 0, sv_uncheck) == CFI_FAILED) {
+            if (fill_shadow_value_to_shadow((uintptr_t)p->map, (uintptr_t)p->map + p->map_len, 0, sv_uncheck) == CFI_FAILED) {
@@ -428,2 +428,2 @@
-            uintptr_t end = p->map + p->map_len;
-            uintptr_t cfi_check = LADDR(cfi_check_sym.dso, cfi_check_sym.sym->st_value);
+            uintptr_t end = (uintptr_t)p->map + p->map_len;
+            uintptr_t cfi_check = (uintptr_t)LADDR(cfi_check_sym.dso, cfi_check_sym.sym->st_value);
@@ -457 +457 @@
-            if (fill_shadow_value_to_shadow(p->map, end, cfi_check, sv_valid_min) == CFI_FAILED) {
+            if (fill_shadow_value_to_shadow((uintptr_t)p->map, end, cfi_check, sv_valid_min) == CFI_FAILED) {
--- a/ldso/dynlink.c
+++ b/ldso/dynlink.c
@@ -580 +580 @@
-		callback(p->map, p->map_len, p->name);
+		callback((uintptr_t)p->map, p->map_len, p->name);
@@ -644 +644 @@
-		callback(p->map, p->map_len, p->name);
+		callback((uintptr_t)p->map, p->map_len, p->name);
@@ -1456 +1456 @@
-			do_pauth_reloc(reloc_addr, base + addend);
+			do_pauth_reloc(reloc_addr, (size_t)base + addend);
@@ -6592 +6592 @@
-		size_t seg_start = p->base + ph->p_vaddr;
+		size_t seg_start = (size_t)p->base + ph->p_vaddr;
@@ -6724 +6724 @@
-	struct dso* p = addr2dso(dso_handle);
+	struct dso* p = addr2dso((size_t)dso_handle);
