diff --git a/sys-devel/autoconf_archive/autoconf_archive-2024.10.16.recipe b/sys-devel/autoconf_archive/autoconf_archive-2024.10.16.recipe
index b9b93e257d..b9b93e257d 100644
--- a/sys-devel/autoconf_archive/autoconf_archive-2024.10.16.recipe
+++ b/sys-devel/autoconf_archive/autoconf_archive-2024.10.16.recipe
@@ -8 +8,2 @@
-SOURCE_URI="http://ftpmirror.gnu.org/autoconf-archive/autoconf-archive-$portVersion.tar.xz"
+SOURCE_URI="http://ftpmirror.gnu.org/autoconf-archive/autoconf-archive-$portVersion.tar.xz
+\thttps://ftp.gnu.org/gnu/autoconf-archive/autoconf-archive-$portVersion.tar.xz"
