diff --git a/build/jam/packages/HaikuBootstrap b/build/jam/packages/HaikuBootstrap
index 2b3d6e44f8..0000000000 100644
--- a/build/jam/packages/HaikuBootstrap
+++ b/build/jam/packages/HaikuBootstrap
@@ -55,6 +55,7 @@ AddFilesToPackage add-ons kernel generic :
-	dpc
-	mpu401
-	scsi_periph
+	dpc
+	mpu401
+	scsi_periph
+	smbios@x86,x86_64
-	<module>tty
-	;
+	<module>tty
+	;
 AddFilesToPackage add-ons kernel partitioning_systems :
