diff --git a/sys-devel/autoconf_archive/autoconf_archive-2024.10.16.recipe b/sys-devel/autoconf_archive/autoconf_archive-2024.10.16.recipe
index b9b93e257d..b9b93e257d 100644
--- a/sys-devel/autoconf_archive/autoconf_archive-2024.10.16.recipe
+++ b/sys-devel/autoconf_archive/autoconf_archive-2024.10.16.recipe
@@ -8 +8,2 @@
-SOURCE_URI="http://ftpmirror.gnu.org/autoconf-archive/autoconf-archive-$portVersion.tar.xz"
+SOURCE_URI="http://ftpmirror.gnu.org/autoconf-archive/autoconf-archive-$portVersion.tar.xz
+	https://ftp.gnu.org/gnu/autoconf-archive/autoconf-archive-$portVersion.tar.xz"

diff --git a/dev-libs/libiconv/libiconv-1.18.recipe b/dev-libs/libiconv/libiconv-1.18.recipe
index 4fb5a4a46a..4fb5a4a46a 100644
--- a/dev-libs/libiconv/libiconv-1.18.recipe
+++ b/dev-libs/libiconv/libiconv-1.18.recipe
@@ -78 +78,2 @@
-SOURCE_URI="https://ftpmirror.gnu.org/libiconv/libiconv-$portVersion.tar.gz"
+SOURCE_URI="https://ftpmirror.gnu.org/libiconv/libiconv-$portVersion.tar.gz
+	https://ftp.gnu.org/gnu/libiconv/libiconv-$portVersion.tar.gz"

diff --git a/dev-libs/expat/expat-2.8.2.recipe b/dev-libs/expat/expat-2.8.2.recipe
index 1f91c5a4c8..1f91c5a4c8 100644
--- a/dev-libs/expat/expat-2.8.2.recipe
+++ b/dev-libs/expat/expat-2.8.2.recipe
@@ -7 +7,2 @@
-SOURCE_URI="https://downloads.sourceforge.net/expat/expat-$portVersion.tar.bz2"
+SOURCE_URI="https://downloads.sourceforge.net/expat/expat-$portVersion.tar.bz2
+	https://github.com/libexpat/libexpat/releases/download/R_${portVersion//./_}/expat-$portVersion.tar.bz2"
